var/spack/repos/builtin/packages/elemental/package.py

```diff
@@ -145,7 +150,16 @@ def cmake_args(self):
             libfortran = LibraryList(mpif77('--print-file-name',
                                             'libgfortran.%s' % dso_suffix,
                                             output=str))
-        if libfortran:
+        elif self.spec.satisfies('%xl') or self.spec.satisfies('%xl_r'):
+            xl_fort = env['SPACK_F77']
+            xl_bin = os.path.dirname(xl_fort)
+            xl_root = os.path.dirname(xl_bin)
+            libfortran = LibraryList('{0}/lib/libxlf90_r.{1}.1'
+                                     .format(xl_root, dso_suffix))
+        else:
+            libfortran = None
+
+        if 'libfortran' in locals():
```

Comment:
@bvanessen Now that `libfortran` is guaranteed to be defined you can ask `if libfortran` - my thought here was that I haven't seen the use of `locals()` before and so it would be more immediately recognizable what was going on if you handled the definition of `libfortran` above and then simply asked `if libfortran` here. (I'm flexible on this point of you have a preference)